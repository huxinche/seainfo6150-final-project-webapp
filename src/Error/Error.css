.not-found{
    text-align: center;
    color: rgb(3, 129, 129);
}

.not-found .large{
    font-size: 90px;
}

@media only screen and (min-width: 768px) {
    .wapper{
        padding: 20px 20px;
    }
    .not-found{
        display: grid;
        grid-template-columns: 0.8fr 1.2fr;
        column-gap: 50px;
        row-gap: 5px;
        justify-items: center;
        align-items: center;
    }
    .not-found .large{
        font-size: 70px;
    }
}

@media screen and (min-width: 992px) {
    .not-found{
        display: grid;
        grid-template-columns: 0.8fr 1.2fr;
        column-gap: 100px;
        row-gap: 10px;
        justify-items: center;
        align-items: center;
       
    }
}


