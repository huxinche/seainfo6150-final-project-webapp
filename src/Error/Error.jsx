import React from 'react'
import { Link } from 'react-router-dom'
import './Error.css'
const Error = () => {
    return (
        <div className="container">
            <div className="not-found">
                <div><p className="center" style={{ textAlign: 'center' }}>
                    <img
                        srcSet={`
                        http://placekitten.com/300/300 300w,
                        http://placekitten.com/400/400 400w,
                        http://placekitten.com/500/500 500w,
                        http://placekitten.com/600/600 600w`}
                        sizes="(max-width: 600px) 400px,
                    (max-width: 768px) 300px,
                    (max-width: 992px) 400px,
                    (max-width: 1200px) 500px,
                    600px"
                        src="http://placekitten.com/200/200"
                        alt="a kitten" />

                </p></div>
                <div>
                    <h1 className="large">404</h1>
                    <h1 className="center">Opps!! <br /> Page Not Found</h1>
                    <br />
                    <br />
                    <div style={{ textAlign: 'center' }}>
                        <Link to="/" className="btn-back">Back to Home</Link>
                    </div>
                    <br />
                    <br />
                </div>

            </div>
        </div>
    )
}

export default Error
