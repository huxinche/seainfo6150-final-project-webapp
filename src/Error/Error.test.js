import React from "react";
import Error from "./Error";

it("renders the Error component correctly", () => {
    
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
})