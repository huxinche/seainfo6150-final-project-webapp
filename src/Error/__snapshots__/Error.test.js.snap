// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="not-found"
    >
      <div>
        <p
          class="center"
          style="text-align: center;"
        >
          <img
            alt="a kitten"
            sizes="(max-width: 600px) 400px, (max-width: 768px) 300px, (max-width: 992px) 400px, (max-width: 1200px) 500px, 600px"
            src="http://placekitten.com/200/200"
            srcset="
                        http://placekitten.com/300/300 300w,
                        http://placekitten.com/400/400 400w,
                        http://placekitten.com/500/500 500w,
                        http://placekitten.com/600/600 600w"
          />
        </p>
      </div>
      <div>
        <h1
          class="large"
        >
          404
        </h1>
        <h1
          class="center"
        >
          Opps!! 
          <br />
           Page Not Found
        </h1>
        <br />
        <br />
        <div
          style="text-align: center;"
        >
          <a
            class="btn-back"
            href="/"
          >
            Back to Home
          </a>
        </div>
        <br />
        <br />
      </div>
    </div>
  </div>
</div>
`;
